### DISCUSSION TITLE
Large document retrieved on PC but not mobile

### DISCUSSION THREAD

**Discussion by mackatap** [GodotNuts/GodotFirebase#406] [Q&A] [answered]
Hello, I am wondering if there is anything in this plugin that would cause my phone to timeout before retrieving a document where my pc doesn't have an issue. I am using a slightly older version of this plugin. If the document is smaller it works on my android, but when it gets bigger, it seems to timeout, and the cache needs to be cleared before I can connect to the database again.  It also only seems to be for a large document that has about 10 fields. a similar size document with only 1 field has no trouble. Thanks!

**Comment by mackatap** [comment]
Also, I think my version doesn't even use offline features so caching isn't helpful. How difficult would it be to tear that stuff out? Caching seems to just cause problems, and I may do it manually.

**Comment by WolfgangSenff** [comment]
What version are you using? Caching should be disabled by default in more recent versions. If you're on 3.x, I guess I didn't do it there though. 4.x, it's definitely disabled. It's easy enough to turn it off without removing it, but in 4.x, I believe I have removed it entirely.

**Comment by WolfgangSenff** [comment]
As for large documents, 10 fields is not that many. How are you using it? Like what's your code?

**Comment by WolfgangSenff** [comment]
Sorry for the multiple posts, but also, I'll be looking into implementing the change that I made on the 4.x side for Firestore recently which will immediately remove the caching automatically as well. I haven't had time for it, and also only seem to be losing motivation instead of gaining it :P, but it's still something which definitely needs to happen. How urgent is your need for that version of the library, if you're on 3.x as I suspect?

**Comment by mackatap** [comment]
![Screenshot 2024-06-18 1645634535308](https://github.com/GodotNuts/GodotFirebase/assets/85116220/c0c0eab7-3bad-4134-9a42-c515b270f368)
Hi, I am still in 3.x. I figured out that the field count doesn't matter, but I have very large maps of arrays in the fields. A map size around 3500 for two of the fields, that are arrays of size 3, whose values are numbers. In all I calculated to be far under the 1mb maximum, more like 250kb. A smaller document of the same format will work, but if it gets up to even 1000 it will timeout and require cache clearing. It does trigger the task_error.

**Reply by mackatap** [reply, marked as answer]
Ok so it was the timeout in the firestore.gd. I increased it to 15 seconds, which seems excessive but ah well it works now and it is just watching leaderboard replays. Not sure why it takes longer on mobile than on pc. I'd still like to disable caching if you could point me in the right direction to do that. Thanks for you help as always!!

**Reply by WolfgangSenff** [reply]
Very interesting! Well, thanks to you, I've started on the big refactor. I'll be removing the problematic parts with caching shortly this morning, after I have taken care of my cats. :) You can follow the updates for it here if you want (this links to a specific article saying I'm starting, but you can join for free, as these updates are always available to free members): https://www.patreon.com/posts/second-big-for-3-106480805?utm_medium=clipboard_copy&utm_source=copyLink&utm_campaign=postshare_creator&utm_content=join_link

I'll leave this discussion open for now and only close it after I've removed the caching mechanism in 3.x! It will probably not take long.

**Comment by WolfgangSenff** [comment]
Okay, caching in Firestore has officially been removed! Are you able to download the current version and test it, or do you need me to make a release, and/or update it in the asset library?

**Comment by WolfgangSenff** [comment]
Alright! The biggest update for 3.x has been pushed. I hope it works for you! This update includes making Firestore be more like 4.x, which is huge. It'll make maintaining the 3.x branch going forward a lot easier. I have NOT updated the wiki yet, I need some time to do that. Feel free to get latest on the main branch and let me know if you like it, and/or if you run into any bugs! That said, it ran through the Test Harness perfectly and worked for every test. The one thing I didn't add from 4.x is the listening endpoint and committing changes. I didn't want that included in the same set of commits. I'm still deciding if I want to add them at all. Committing changes would give you the ability to have server-side timestamps and a few other very small things, let me know if that's interesting to you or no. I'm going to go ahead and close this now, as the issue is resolved quite well! (By the way, the thing will be cached, but in a node in your scene tree, meaning you can just pass true/false to get_doc - the newly named get function - as the second parameter and if you pass true, it won't even bother to send a request. Useful if you know your app doesn't need an updated copy of the document, and if you don't want to have to save the document in a singleton across scenes. Plus, if you pass true and the document isn't cached in the tree, it'll actually just make a request to go get it. I didn't make true the default though to maintain backward compatibility. In any case, you can let me know if these are useful at patreon.com/KyleSzklenski - probably easier than here, and I support free memberships!

**Reply by mackatap** [reply]
Thank you very much for your work!!! I was about to implement my own caching system for certain documents but I will go ahead and try out this update. I am a bit nervous as I have chopped up a good deal of this addon code over the years to get around odd things I have done. I am on a quite old version now but would love to be back on a newer version.

**Comment by WolfgangSenff** [comment]
Well make sure to save your current version just in case! 🤣
